src/main/java/com/salesforce/storm/spout/sideline/trigger/example/ZookeeperWatchTrigger.java

```diff
@@ -0,0 +1,374 @@
+        try {
+            final FilterChainStep step = filterChainStepBuilder.build(triggerEvent.getData());
+
+            final SidelineRequest sidelineRequest = new SidelineRequest(
+                generateSidelineRequestIdentifier(triggerEvent, step),
+                step
+            );
+
+            logger.info("Creating a sideline request with id {} and step {}", sidelineRequest.id, sidelineRequest.step);
+
+            return sidelineRequest;
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException ex) {
+            logger.error("Unable to generate an identifier for this request, cowardly refusing to proceed! {}", ex);
+            return null;
+        }
+    }
+
+    /**
+     * Using a filter chain step, make some JSON out of it and then hash it to create an idempotent identifier.
+     * @param triggerEvent Trigger event that contains metadata about the request
+     * @param step The FilterChainStep that we are going to generate the request from
+     * @return A sideline request identifier for the filter chain step
+     * @throws NoSuchAlgorithmException Your java install is whack yo, it's missing MD5, for realz???
+     */
+    private SidelineRequestIdentifier generateSidelineRequestIdentifier(
+        final TriggerEvent triggerEvent,
+        final FilterChainStep step
+    ) throws NoSuchAlgorithmException, UnsupportedEncodingException {
+        final String json = gson.toJson(step);
+
+        final StringBuilder identifier = new StringBuilder(
+            DatatypeConverter.printHexBinary(MessageDigest.getInstance("MD5").digest(json.getBytes("UTF-8")))
+        );
+
+        // If we were provided a date time in the event, append the time stamp of that event to the identifier
+        if (triggerEvent.getCreatedAt() != null) {
+            identifier.append("-");
+            identifier.append(
+                triggerEvent.getCreatedAt().getTime()
+            );
+        }
+
+        return new SidelineRequestIdentifier(identifier.toString());
+    }
+
+    /**
+     * Create a trigger event from the provided data.
+     * @return Trigger event
+     */
+    private TriggerEvent getTriggerEvent(final byte[] data) {
+        final String json = new String(data, Charset.forName("UTF-8"));
+        return getTriggerEventFromJson(json);
+    }
+
+    /**
+     * Parse a trigger event from some JSON.
+     * @param json JSON to parse
+     * @return Trigger event
+     */
+    private TriggerEvent getTriggerEventFromJson(final String json) {
+        try {
+            return gson.fromJson(json, TriggerEvent.class);
+        } catch (Exception e) {
+            logger.error("Unable to parse trigger event {} {}", json, e);
+            return null;
+        }
+    }
+
+    /**
+     * Watch implementation for the sideline trigger node in Zookeeper.
+     */
+    private class SidelineTriggerWatch implements PathChildrenCacheListener {
```

Comment:
Intentionally not making this static, so I can easily reference methods on the outer class.